### ISSUE TITLE
retrain im2txt on custom dataset

### ISSUE THREAD

**Issue by ZixuanLiang** [tensorflow/models#2353]
@cshallue I built my own TFRecord files according to **build_mscoco_data.py** except I didn't add start word < S > and end word < /S > to captions. Then in **configuration.py**, I adjusted **vocab_size**, **values_per_input_shard**, **num_examples_per_epoch** according to my dataset. When I started training, the loss was always 0.0000. I think there was something wrong. What did I miss? 

2017-09-07 02:23:40.321952: I tensorflow/core/common_runtime/gpu/gpu_device.cc:908] DMA: 0 
2017-09-07 02:23:40.321958: I tensorflow/core/common_runtime/gpu/gpu_device.cc:918] 0:   Y 
2017-09-07 02:23:40.321968: I tensorflow/core/common_runtime/gpu/gpu_device.cc:977] Creating TensorFlow device (/gpu:0) -> (device: 0, name: Tesla K80, pci bus id: 0000:00:1e.0)
INFO:tensorflow:Restoring Inception variables from checkpoint file /data/upload/inception_v3.ckpt
INFO:tensorflow:Restoring parameters from /data/upload/inception_v3.ckpt
INFO:tensorflow:Starting Session.
INFO:tensorflow:Starting Queues.
INFO:tensorflow:global_step/sec: 0
INFO:tensorflow:global step 1: loss = 9.0076 (23.07 sec/step)
INFO:tensorflow:global step 2: loss = 0.1956 (0.59 sec/step)
2017-09-07 02:24:07.753348: I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 6414 get requests, put_count=6168 evicted_count=1000 eviction_rate=0.162127 and unsatisfied allocation rate=0.209853
2017-09-07 02:24:07.753407: I tensorflow/core/common_runtime/gpu/pool_allocator.cc:259] Raising pool_size_limit_ from 100 to 110
INFO:tensorflow:global step 3: loss = 0.0000 (0.64 sec/step)
INFO:tensorflow:global step 4: loss = 0.0000 (0.62 sec/step)
INFO:tensorflow:global step 5: loss = 0.0000 (0.64 sec/step)
INFO:tensorflow:global step 6: loss = 0.0000 (0.60 sec/step)
INFO:tensorflow:global step 7: loss = 0.0000 (0.66 sec/step)
INFO:tensorflow:global step 8: loss = 0.0000 (0.61 sec/step)
INFO:tensorflow:global step 9: loss = 0.0000 (0.60 sec/step)
INFO:tensorflow:global step 10: loss = 0.0000 (0.61 sec/step)
INFO:tensorflow:global step 11: loss = 0.0000 (0.63 sec/step)
INFO:tensorflow:global step 12: loss = 0.0000 (0.64 sec/step)
INFO:tensorflow:global step 13: loss = 0.0000 (0.69 sec/step)
INFO:tensorflow:global step 14: loss = 0.0000 (0.61 sec/step)
INFO:tensorflow:global step 15: loss = 0.0000 (0.66 sec/step)
INFO:tensorflow:global step 16: loss = 0.0000 (0.62 sec/step)
INFO:tensorflow:global step 17: loss = 0.0000 (0.61 sec/step)
INFO:tensorflow:global step 18: loss = 0.0000 (0.63 sec/step)
INFO:tensorflow:global step 19: loss = 0.0000 (0.70 sec/step)
INFO:tensorflow:global step 20: loss = 0.0000 (0.60 sec/step)
INFO:tensorflow:global step 21: loss = 0.0000 (0.62 sec/step)
INFO:tensorflow:global step 22: loss = 0.0000 (0.61 sec/step)
INFO:tensorflow:global step 23: loss = 0.0000 (0.63 sec/step)
INFO:tensorflow:global step 24: loss = 0.0000 (0.64 sec/step)
INFO:tensorflow:global step 25: loss = 0.0000 (0.61 sec/step)

**Comment by ZixuanLiang** [comment]
I added start word and end word. It is working now. But I don't know how start word and end word are used during training. Could someone explain to me?

**Comment by reedwm** [comment]
This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!
